lib/cisco_node_utils/yum.rb

```diff
@@ -66,27 +76,73 @@ def self.detect_vrf
     end
 
     def self.install(pkg, vrf=nil)
-      vrf = vrf.nil? ? detect_vrf : "vrf #{vrf}"
-      config_set('yum', 'install', pkg, vrf)
+      if platform == :nexus
+        vrf = vrf.nil? ? detect_vrf : "vrf #{vrf}"
+        config_set('yum', 'install', pkg, vrf)
 
-      # HACK: The current nxos host installer is a multi-part command
-      # which may fail at a later stage yet return a false positive;
-      # therefore a post-validation check is needed here to verify the
-      # actual outcome.
+      elsif platform == :ios_xr
+        filename = pkg.strip.tr(':', '/').split('/').last
+        pkg_info = Yum.decompose_name(filename)
+        if pkg_info
+          pkg_name = "#{pkg_info[1]}-#{pkg_info[2]}-#{pkg_info[3]}"
+          debug "Installing package #{pkg_name}"
+          `#{EXEC_IN_DEFAULT_NS} install activate pkg 0x0 #{pkg_name}`
+          `#{EXEC_IN_DEFAULT_NS} install commit sdr`
+        else
+          fail 'Failed to install the requested rpm.'
```

Comment:
Can you add more context to this failure message? Looks like you get here if you're not going to try to install it (because the decompose_name call failed), not because installation failed.